chcemvediet/apps/inforequests/tests/test_views/test_mine.py

```diff
@@ -0,0 +1,84 @@
+# vim: expandtab
+# -*- coding: utf-8 -*-
+from django.test import TestCase
+
+from poleno.utils.test import ViewTestCaseMixin
+from poleno.utils.urls import reverse
+
+from .. import InforequestsTestCaseMixin
+from ...models import Action
+
+
+class MineViewTest(InforequestsTestCaseMixin, ViewTestCaseMixin, TestCase):
+    u"""
+    Tests ``mine()`` view registered as "inforequests:mine".
+    """
+
+    def _create_successful_inforequest(self, applicant):
+        inforequest, _, _ = self._create_inforequest_scenario(applicant, {u'closed': True},
+                (u'disclosure', dict(disclosure_level=Action.DISCLOSURE_LEVELS.PARTIAL)))
+        return inforequest
+
+
+    def test_allowed_http_methods(self):
+        allowed = [u'HEAD', u'GET']
+        self.assert_allowed_http_methods(allowed, reverse(u'inforequests:mine'))
+
+    def test_anonymous_user_is_redirected(self):
+        self.assert_anonymous_user_is_redirected(reverse(u'inforequests:mine'))
+
+    def test_authenticated_user_gets_inforequest_mine(self):
+        self._login_user(self.user1)
```

Comment:
@martinmacko47 Explicitne som tu uviedol `self.user1`. Defaultna hodnota `self.user` je vytvorena v `ChcemvedietTestCaseMixin` ale s inym hashovanim ako je MD5, preto bol `self.login()` neuspesny. Chcel som teda prepisat settingy aj v `ChcemvedietTestCaseMixin` ale narazil som n a maximum recursion error pri nested voladi `self._postTearDown()`. Asi podobny problem [sem](https://code.djangoproject.com/ticket/20290) 
"Refactored override_settings to store the underlying settings._wrapped value seen at runtime, not instantiation time."
Problem nenastal ak sa premenna `self.settings_override` volala v oboch classach rozne. Nakoniec som vsak pre login puzil `self.user1` ktory je definovany v `InforequestTestCaseMixin` s MD5 hashovanim.

Reply:
*(na toto sa pozriem, az ked vyriesime, co som pisal v ostatnych dvoch komentaroch)*

Reply:
Je to kvoli funkcii `self._login`. Vsetky inforequest view-testy, vsade kde sa pouziva dana funkcia, su preskocene. Napr. `skip_test_create` test `test_user_with_verified_email_gets_inforequest_create` vrati status code **302** (namiesto ocakavanych **200**), `self.user` sa neprihlasil. Ak som pouzil `self._login_user(self.user1)` tak sa user uspesne prihlasil. 

Z tohto dovodu by sme mohli upravit metodu aby namiesto `None` vratilo boolean, ci prihlasenie bolo uspesne alebo neuspesne a v teste to aj assertovat, dokazame potom skor najst chybu preco test padol.

Reply:
```
self._login_user()  # False
self._login_user(self.user1)  # True
```

Reply:
Ak by sme odstranili `PASSWORD_HASHERS` z `InforequestsTestCaseMixin`, tak by login fungoval spravne 
```
self._login_user()  # True
self._login_user(self.user1)  # True
```

Cas potrebny na zbehnutie testov v inforequest appke  bez / s MD5-hashovanim je 84 / 24 sec, vsetky testy v projekte 159 / 63 sec, takze vyrazna casova uspora. Nastavil som teda MD5 hashovanie pre vsetky testy do `CustomTestRunner`.

Reply:
> Z tohto dovodu by sme mohli upravit metodu aby namiesto None vratilo boolean, ci prihlasenie bolo uspesne alebo neuspesne a v teste to aj assertovat, dokazame potom skor najst chybu preco test padol.

Skor by som ju pre jednoduchost upravil tak, ze ak sa login nepodari, tak rovno raisne assert expection. Kedze vzdy ocakavame, ze login sa ma podarit. Neviem ako sa da zistit, ci sa login podaril. Ale rovnakym sposobom ako si planoval vratit `True`/`False` by sme rovno assertli, ze je to vzdy `True`.

Reply:
> Cas potrebny na zbehnutie testov v inforequest appke bez / s MD5-hashovanim je 84 / 24 sec, vsetky testy v projekte 159 / 63 sec, takze vyrazna casova uspora. Nastavil som teda MD5 hashovanie pre vsetky testy do `CustomTestRunner`.

Super. Ak by sme predsa len niekde potrebovali skutocny hash, tak tam to lokalne overridneme a vytvorime usera s inym hashom. Ale nenapada mi test, kde by sme potrebovali testovat nieco specialne s hashom.

Reply:
:ok:, `self.client.login` vrati  boolean. 
"""
Returns True if login is possible; False if the provided credentials
        are incorrect, or the user is inactive, or if the sessions framework is
        not available.
"""